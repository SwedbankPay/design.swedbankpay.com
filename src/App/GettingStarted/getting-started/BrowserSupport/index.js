import React from "react";

import { DocContainer } from "@docutils";

import Chrome from "./browser-icons/chrome.svg";
import Firefox from "./browser-icons/firefox.svg";
import Edge from "./browser-icons/edge.svg";
import InternetExplorer from "./browser-icons/internet-explorer.svg";
import Opera from "./browser-icons/opera.svg";
import Safari from "./browser-icons/safari.svg";

const BrowserSupport = () => (
    <DocContainer>
        {/* TODO: Replace this with styled components [EH] */}
        <style dangerouslySetInnerHTML={{ __html: `
            .browser-icon {
                height: 48px;
                width: 48px;
            }
        ` }} />
        <p className="lead">Currently supported browsers:</p>
        <table className="browser-support-table table">
            <thead>
                <tr>
                    <th>
                        <img className="browser-icon" src={Chrome} alt="Google Chrome Icon" />
                    </th>
                    <th>
                        <img className="browser-icon" src={Firefox} alt="Mozilla Firefox Icon" />
                    </th>
                    <th>
                        <img className="browser-icon" src={Edge} alt="Microsoft Edge Icon"/>
                    </th>
                    <th>
                        <img className="browser-icon" src={InternetExplorer} alt="Microsoft Internet Explorer Icon"/>
                    </th>
                    <th>
                        <img className="browser-icon" src={Opera} alt="Opera Browser Icon"/>
                    </th>
                    <th>
                        <img className="browser-icon" src={Safari} alt="Apple Safari Icon"/>
                    </th>
                </tr>
            </thead>
            <tbody>
                <tr>
                    <td className="background-brand">Yes</td>
                    <td className="background-warning">Yes</td>
                    <td className="background-warning">Yes</td>
                    <td className="background-warning">11</td>
                    <td className="background-warning">Yes</td>
                    <td className="background-warning">Yes</td>
                </tr>
            </tbody>
        </table>
    </DocContainer>
);

export default BrowserSupport;
