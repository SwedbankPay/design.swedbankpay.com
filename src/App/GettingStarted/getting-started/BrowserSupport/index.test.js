import React from "react";
import { shallow } from "enzyme";

import BrowserSupport from "./index";

describe("GettingStarted: BrowserSupport", () => {
    it("is defined", () => {
        expect(BrowserSupport).toBeDefined();
    });

    it("renders", () => {
        const wrapper = shallow(<BrowserSupport />);

        expect(wrapper).toMatchSnapshot();
    });
});
