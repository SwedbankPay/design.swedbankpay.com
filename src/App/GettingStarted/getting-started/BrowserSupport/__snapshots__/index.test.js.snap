// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GettingStarted: BrowserSupport renders 1`] = `
<DocContainer>
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": "
            .browser-icon {
                height: 48px;
                width: 48px;
            }
        ",
      }
    }
  />
  <p
    className="lead"
  >
    Currently supported browsers:
  </p>
  <table
    className="browser-support-table table"
  >
    <thead>
      <tr>
        <th>
          <img
            alt="Google Chrome Icon"
            className="browser-icon"
            src=""
          />
        </th>
        <th>
          <img
            alt="Mozilla Firefox Icon"
            className="browser-icon"
            src=""
          />
        </th>
        <th>
          <img
            alt="Microsoft Edge Icon"
            className="browser-icon"
            src=""
          />
        </th>
        <th>
          <img
            alt="Microsoft Internet Explorer Icon"
            className="browser-icon"
            src=""
          />
        </th>
        <th>
          <img
            alt="Opera Browser Icon"
            className="browser-icon"
            src=""
          />
        </th>
        <th>
          <img
            alt="Apple Safari Icon"
            className="browser-icon"
            src=""
          />
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          className="background-brand"
        >
          Yes
        </td>
        <td
          className="background-warning"
        >
          Yes
        </td>
        <td
          className="background-warning"
        >
          Yes
        </td>
        <td
          className="background-warning"
        >
          11
        </td>
        <td
          className="background-warning"
        >
          Yes
        </td>
        <td
          className="background-warning"
        >
          Yes
        </td>
      </tr>
    </tbody>
  </table>
</DocContainer>
`;
